src/atoms/Radio/Radio.web.js

```diff
@@ -1,5 +1,50 @@
-import React from 'react';
+import React, { useState, useCallback, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { getVariantColorKeys } from '../../_helpers/theme';
+import RadioButtonContext from './RadioContext';
+import RadioOption from './RadioOption';
+import Fieldset from './Fieldset';
 
-const RadioButton = () => <div>WIP</div>;
+const Radio = ({ value, onChange, defaultValue, variantColor, size, children }) => {
+  const [selected, setSelected] = useState(value || defaultValue);
 
-export default RadioButton;
+  const onValueChange = useCallback(
+    (newValue) => {
+      if (!value) {
+        setSelected(newValue);
+      }
+      onChange(newValue);
```

Comment:
Why didn't we put this into the `if` block itself?

Reply:
controlled/uncontrolled differences. If `value` was passed means its controlled component, so dont set its own state.
Irrespective of it onChange is still called.